routing/pathfind_test.go

```diff
@@ -2010,3 +2010,99 @@ func TestRestrictOutgoingChannel(t *testing.T) {
 			"but channel %v was selected instead", route.Hops[0].ChannelID)
 	}
 }
+
+// TestCltvLimit asserts that a cltv limit is obeyed by the path finding
+// algorithm.
+func TestCltvLimit(t *testing.T) {
+	t.Run("no limit", func(t *testing.T) { testCltvLimit(t, 0, 1) })
+	t.Run("no path", func(t *testing.T) { testCltvLimit(t, 50, 0) })
+	t.Run("force high cost", func(t *testing.T) { testCltvLimit(t, 80, 2) })
+}
+
+func testCltvLimit(t *testing.T, limit uint32, expectedChannel uint64) {
+	t.Parallel()
+
+	// Set up a test graph with two possible paths to the target. The path
+	// through channel 2 is the highest cost path, but has the lowest cltv
+	// delta.
+	testChannels := []*testChannel{
```

Comment:
Would that test the cltv limit? If I understand you correctly, that would test another part of path finding: the fee function and the trade off between fees and time lock?

Reply:
> Would that test the cltv limit?

Correct.

> If I understand you correctly, that would test another part of path finding: the fee function and the trade off between fees and time lock?

I would say this would test the tradeoff between path length and CLTV limit. Two paths (one longer than the other) have the same fee cost, but the shorter path has a CLTV higher than the limit, so the longer path should be chosen.

Reply:
Got it. Updated.